### PR TITLE
KEYCLOAK-11874 User should be evicted from cache when their credential is deleted

### PR DESCRIPTION
**Pull request by tkyjovsk**
- added eviction of user after their credential was deleted
- added `UserTest.loginShouldFailAfterPasswordDeleted()`

### CODE REVIEW

**Comment by AlistairDoswald** [comment]
@tkyjovsk Following this error, I went though the code to look for other locations where this could be a problem. I think we should also evict the cache in PasswordCredentialProvider.deleteCredential(RealmModel realm, UserModel user, String credentialId) (even though currently it's never called). Could you also do it it this PR? If not I'll just validate this PR, and do a follow-up one.

**Comment by tkyjovsk** [comment]
@AlistairDoswald Thinking about it, perhaps the eviction should be done after any credential deletion of any type, not just the passwords. In that case perhaps the `UserCredentialStoreManager.removeStoredCredential()` would be a better place to put it? That method is being called from all `*CredentialProvider.deleteCredential()` methods as well as the `UserResource.removeCredential()` method.

**Comment by AlistairDoswald** [comment]
@tkyjovsk yes, it would be better I think. Safer at the very least.

**Comment by tkyjovsk** [comment]
So I moved the eviction call from `UserResource` to the `UserCredentialStoreManager.removeStoredCredential()`. The test case is passing.

**Review by mposolda — APPROVED** [review]
@tkyjovsk @AlistairDoswald Agree about UserCredentialStoreManager being safer option. Approving.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java [hunk 1]
```diff
@@ -1556,4 +1556,31 @@ private void enableBruteForce(boolean enable) {
         assertAdminEvents.assertEvent(realmId, OperationType.UPDATE, Matchers.nullValue(String.class), rep, ResourceType.REALM);
     }
 
+    @Test
+    public void loginShouldFailAfterPasswordDeleted() {
+        String userName = "credential-tester";
```
Comment (AlistairDoswald) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java:1561]: User `credential-tester` isn't getting cleaned up correctly after this test, and is causing tests count() and countUsersNotServiceAccount() to fail.

**Review by AlistairDoswald — CHANGES_REQUESTED** [review]
Small clean-up problem introduced by this test. I'll be able to merge after it is corrected.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java (hunk 1 above)
Comment (tkyjovsk) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java:1561]: Right, sorry for missing that. It should be fixed in the next push.

**Review by AlistairDoswald — APPROVED** [review]

**Pull request merged**